railties/lib/rails/generators/app_base.rb

```diff
@@ -539,6 +539,11 @@ def dockerfile_yarn_version
         "latest"
       end
 
+      def yarn_through_corepack?
+        true if dockerfile_yarn_version == "latest"
+        dockerfile_yarn_version.to_s.split(".").first.to_i >= 2
```

Comment:
```suggestion
        dockerfile_yarn_version >= "2"
```

Reply:
Thanks @zzak, I didn't know we can do that in Ruby tricks!
Thank you for the learning!